package.json

```diff
@@ -25,10 +25,11 @@
     "lint": "eslint */**/**/*.{js,ts,css,json,md} --fix --ext",
     "watch": "npm run build -- -w",
     "compile": "tsc -b ./server ./clients/vscode",
-    "publish": "npm run build && lerna publish",
-    "postpublish": "(cd ./clients/vscode && npm i && vsce publish)",
+    "vscode:prepublish": "npm run build",
+    "vsce:publish": "sh scripts/publish.sh",
+    "vsce:postpublish": "(cd ./clients/vscode && npm i && sh scripts/publish.sh)",
     "postinstall": "lerna exec npm install",
-    "vsce:check": "yarn shell:lint && sh scripts/check-update.sh",
+    "vsce:check": "yarn shell:lint && sh scripts/check-update.sh && cd ./clients/vscode && yarn shell:lint && sh scripts/check-update.sh",
```

Comment:
It shouldn't be required to call either `shell:lint` or `check-update.sh` multiple times. 

`yarn shell:lint` should be adjusted to test all relevant scripts. Currently, it only looks for `.sh` files in `scripts/` folder. 

`check-update.sh` should be aware of what all files it needs to update and calling it only once should suffice.